src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion_img2img.py

```diff
@@ -178,14 +179,19 @@ def _generate(
         prompt_ids: jnp.array,
         image: jnp.array,
         params: Union[Dict, FrozenDict],
-        prng_seed: jax.random.PRNGKey,
-        strength: float = 0.8,
```

Comment:
Ideally, we could make it so that `strength` can be a sharded array too (behaving exactly like `guidance_scale`) so we can perform multiple parallel requests with different strengths. This would be useful for batching requests, but we can address it in a future PR.

Reply:
Initially I also wanted to do that, but AFIK its not possible (in an easy way):

The strength influences the `start_timestep` which is used in the `jax.lax.fori_loop` bounds. The bounds for the loop must be known at compile time (as with `num_inference_steps`) so AFAIK there is no way to have a different strength per shard in pmap.

The only way I could think of doing that is passing in a static `min_start_timestep` and using that as the loop bounds and then also passing in the dynamic sharded `start_timestep`. Then in the loop body, using a jax conditional discard / "skip" the body if `timestemp < start_timestep` but that would be overkill in my opinion.

Reply:
Btw, the reason I did the `strength` -> `start_timestep` calculation outside the jit is to reduce the number of recompilations.
E.g. `strength(0.8) != strength(0.81)` but `start_timestep(0.8) == start_timestep(0.81)`

Reply:
Agree with @skirsten, `strength` is used to determine the value of inference steps, and different `strength = different number of timesteps`, so it would be hard to parallelize it.

Reply:
Yes, good points @skirsten and @patil-suraj, let's leave it like this.